cypress/e2e/widgets-landing.cy.ts

```diff
@@ -44,4 +88,38 @@ describe('Widget Landing Page', () => {
     // Reset to default layout
     resetToDefaultLayout();
   });
+
+  describe('Widget Layout', () => {
+    beforeEach(() => {
+      cy.login();
+      cy.visit('/');
+      cy.viewport(1280, 2000);
+      cy.get('.react-grid-item').should('be.visible');
+    });
+
+    it('widgets can be dragged and dropped', () => {
+      cy.intercept(
+        'GET',
+        '**/api/chrome-service/v1/dashboard-templates?dashboard=landingPage'
+      ).as('resetLayout');
+      cy.intercept(
+        'PATCH',
+        '**/api/chrome-service/v1/dashboard-templates/*'
+      ).as('patchLayout');
+
+      resetToDefaultLayout();
+      cy.wait('@resetLayout').its('response.statusCode').should('eq', 200);
+      moveWidget(0, 1);
+      cy.wait('@patchLayout').its('response.statusCode').should('eq', 200);
+      getLayout().then((firstMove) => {
```

Comment:
We should use interceptors rather than directly calling API. Users will never interact with the UI like this. If we can't verify the code via the UI, the test does not really achieve anything.

I think you are having issues with the fact that the API calls are denounced. You can wait for a set amount of time in cypress, if you have to wait some API calls but they are delay for what ever reason.

Reply:
Good point. I removed the `getLayout` function and am now using interceptors to check that the layout has been updated. Let me know what you think.

Reply:
Way better :)